test/integration/addons_test.go

```diff
@@ -46,17 +46,20 @@ func TestAddons(t *testing.T) {
 	ctx, cancel := context.WithTimeout(context.Background(), Minutes(40))
 	defer Cleanup(t, profile, cancel)
 
-	// Set an env var to point to our dummy credentials file
-	err := os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", filepath.Join(*testdataDir, "gcp-creds.json"))
-	defer os.Unsetenv("GOOGLE_APPLICATION_CREDENTIALS")
-	if err != nil {
-		t.Fatalf("Failed setting GOOGLE_APPLICATION_CREDENTIALS env var: %v", err)
-	}
+	// We don't need a dummy file is we're on GCE
+	if !detect.IsOnGCE() {
+		// Set an env var to point to our dummy credentials file
+		err := os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", filepath.Join(*testdataDir, "gcp-creds.json"))
+		defer os.Unsetenv("GOOGLE_APPLICATION_CREDENTIALS")
+		if err != nil {
+			t.Fatalf("Failed setting GOOGLE_APPLICATION_CREDENTIALS env var: %v", err)
+		}
 
-	err = os.Setenv("GOOGLE_CLOUD_PROJECT", "this_is_fake")
-	defer os.Unsetenv("GOOGLE_CLOUD_PROJECT")
-	if err != nil {
-		t.Fatalf("Failed setting GOOGLE_CLOUD_PROJECT env var: %v", err)
+		err = os.Setenv("GOOGLE_CLOUD_PROJECT", "this_is_fake")
```

Comment:
isn't this overridng the preivous os.SetEnv ? did u mean to be in a different block ?

Reply:
i think the diff is a little confusing, but this is doing what it always has been, just inside of an if statement